pandas/tests/base/test_misc.py

```diff
@@ -150,6 +151,8 @@ def test_access_by_position(index):
     assert index[-1] == index[size - 1]
 
     msg = f"index {size} is out of bounds for axis 0 with size {size}"
+
+    assert not is_dtype_equal(index.dtype, "string[pyarrow]")
```

Comment:
can you move to the tests for is_dtype_equal,

Reply:
Done